### PR TITLE
Ports asset changes from helius-labs

### PR DESCRIPTION
**Pull request by linuskendall**
Fixes ported from
- fix(ingester): index token freeze
- Backport of https://github.com/helius-labs/digital-asset-rpc-infrastructure/commit/f057cd5c430e8f8ae7528af90ae03ad9696ea20d

### CODE REVIEW

**Comment by danenbm** [comment]
On hold as we expect this PR to be merged as part of a creator/data hash fix PR.

File: nft_ingester/src/program_transformers/token_metadata/v1_asset.rs
Comment (danenbm) [inline nft_ingester/src/program_transformers/token_metadata/v1_asset.rs:109]: Looks like `OwnerType::Single` is for NFTs and `OwnerType::Token` is for fungible assets and fungible tokens.  But wouldn't those also have the same kind of `token_account` to parse as with NFTs, just with supply > 1 and/or decimals > 0?

**Review by danenbm — COMMENTED** [review]
I just have the one question above, otherwise this LGTM

File: nft_ingester/src/program_transformers/token_metadata/v1_asset.rs
Comment (NicolasPennie) [reply nft_ingester/src/program_transformers/token_metadata/v1_asset.rs:103]: You'll have N token_accounts. So you can't find one.

**Review by NicolasPennie — APPROVED** [review]

File: nft_ingester/src/program_transformers/token_metadata/v1_asset.rs
Comment (danenbm) [reply nft_ingester/src/program_transformers/token_metadata/v1_asset.rs:103]: Oh yeah that makes sense I didn't think it through properly.

**Review by danenbm — APPROVED** [review]

**Pull request merged**
